backend/netstorage/netstorage.go

```diff
@@ -0,0 +1,1369 @@
+		err = errors.New("NewFS: " + startURL + " unsupported type: " + f.filetype)
+		f.filetype = ""
+		return err
+	}
+}
+
+// Join's the remote onto the endpoint URL
+func (f *Fs) url(remote string) string {
+	if remote == "" {
+		return f.endpointURL
+	}
+	pathescapeURL := rest.URLPathEscape(remote)
+	//rclone likes to add ./ to the URL for no reason
+	pathescapeURL = strings.TrimPrefix(pathescapeURL, "./")
+	//case happens if we do copy/move file to target non-existent dir with no '/'
+	if f.filetype == "" && remote != "" && !strings.HasSuffix(f.endpointURL, "/") {
+		return f.endpointURL + "/" + pathescapeURL
+	}
+	//this should't happen if URLs are properly checked/formatted
+	if !strings.HasSuffix(f.endpointURL, "/") && pathescapeURL != "" {
+		fs.Errorf(nil, "joining endpoint without '/': %s to remote: %s", f.endpointURL, pathescapeURL)
+		debug.PrintStack()
+	}
+	return f.endpointURL + pathescapeURL
+}
+
+// Adds the configured headers to the request if any
+func (f *Fs) addHeaders(req *http.Request, path string, actionHeader string) {
+	dataHeader := generateDataHeader(f)
+	req.Header.Add("X-Akamai-ACS-Auth-Data", dataHeader)
+	req.Header.Add("X-Akamai-ACS-Auth-Sign", generateSignHeader(f, dataHeader, path, actionHeader))
+	req.Header.Add("X-Akamai-ACS-Action", actionHeader)
+}
+
+// getFileName returns the file name if present, otherwise decoded name_base64
+// if present, otherwise an empty string
+func (f *Fs) getFileName(file *File) string {
+	if file.Name != "" {
+		return file.Name
+	}
+	if file.NameBase64 != "" {
+		decoded, err := base64.StdEncoding.DecodeString(file.NameBase64)
+		if err == nil {
+			return string(decoded)
+		}
+		fs.Errorf(nil, "List base64 decode object name %s error: %s", file.NameBase64, err)
+	}
+	return ""
+}
+
+// List the objects and directories in dir into entries.  The
+// entries can be returned in any order but should be for a
+// complete directory.
+//
+// dir should be "" to list the root, and should not have
+// trailing slashes.
+//
+// This should return ErrDirNotFound if the directory isn't
+// found.
+func (f *Fs) List(ctx context.Context, dir string) (entries fs.DirEntries, err error) {
+	if f.filetype == "" {
+		// This happens in two scenarios.
+		// 1. NewFs is done on a non-existent object, then later rclone attempts to List/ListR this NewFs.
+		// 2. List/ListR is called from the context of test_all and not the regular rclone binary.
+		err := f.initFs(ctx, dir)
+		if err != nil {
+			if err == fs.ErrorObjectNotFound {
+				fs.Debugf(nil, "List on dir not found: %s", f.url(dir))
+				return nil, fs.ErrorDirNotFound
+			}
+			return nil, err
+		}
+	}
+
+	URL := f.url(dir)
+	if dir != "" && !strings.HasSuffix(dir, "/") {
+		dir += "/"
+	}
+	files, err := f.netStorageDirRequest(ctx, URL)
+	if err != nil {
+		return nil, err
+	}
+	for _, item := range files {
+		name := dir + f.getFileName(&item)
+		switch item.Type {
+		case "dir":
+			mtime, _ := strconv.ParseInt(item.Mtime, 10, 64)
```

Comment:
Please don't ignore errors!

Reply:
Acually if it is an int, shouldn't you change the definition here?

```
	Mtime      int64   `xml:"mtime,attr"`
```

Same with size, filecount etc

```